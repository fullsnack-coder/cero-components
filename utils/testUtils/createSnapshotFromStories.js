import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
