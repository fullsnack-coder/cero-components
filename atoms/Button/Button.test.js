import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import createSnapshotFromStories from '../../utils/testUtils/createSnapshotFromStories'

import * as stories from './Button.stories'

import Button from './Button'

describe('[ atoms / Button ]', () => {
  describe('#Button', () => {
    // Snapshots based on stories
    createSnapshotFromStories('Button', stories)

    describe('when `Button`is clicked', () => {
      it('should call `onClick`', () => {
        // Arrange
        const onClickMock = jest.fn()
        const props = {
          onClick: onClickMock,
        }

        // Act
        render(<Button {...props}>Button</Button>)
        fireEvent.click(screen.getByText('Button'))

        // Assert
        expect(onClickMock).toHaveBeenCalled()
      })
    })
  })
})
