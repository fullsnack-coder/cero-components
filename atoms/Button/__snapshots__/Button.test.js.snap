// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ atoms / Button ] #Button when 'Button' AddonAppend is mounted should render 1`] = `
<DocumentFragment>
  <button
    class="button type-primary"
  >
    <h1
      class="heading color-inverted size-md weight-bold is-centered is-inline"
    >
      You Shall Not Pass!
    </h1>
    But ill-tempered Thengel.
  </button>
</DocumentFragment>
`;

exports[`[ atoms / Button ] #Button when 'Button' AddonPrepend is mounted should render 1`] = `
<DocumentFragment>
  <button
    class="button type-primary"
  >
    Dare Erebor forgive most.
    <h1
      class="heading color-inverted size-md weight-bold is-centered is-inline"
    >
      You Shall Not Pass!
    </h1>
  </button>
</DocumentFragment>
`;

exports[`[ atoms / Button ] #Button when 'Button' Default is mounted should render 1`] = `
<DocumentFragment>
  <button
    class="button type-primary"
  >
    <h1
      class="heading color-inverted size-md weight-bold is-centered"
    >
      You Shall Not Pass!
    </h1>
  </button>
</DocumentFragment>
`;

exports[`[ atoms / Button ] #Button when 'Button' Inline is mounted should render 1`] = `
<DocumentFragment>
  <button
    class="button type-primary is-inline"
  >
    <h1
      class="heading color-inverted size-md weight-bold is-centered"
    >
      You Shall Not Pass!
    </h1>
  </button>
</DocumentFragment>
`;

exports[`[ atoms / Button ] #Button when 'Button' Muted is mounted should render 1`] = `
<DocumentFragment>
  <button
    class="button type-primary is-muted"
  >
    <h1
      class="heading color-inverted size-md weight-bold is-centered"
    >
      You Shall Not Pass!
    </h1>
  </button>
</DocumentFragment>
`;

exports[`[ atoms / Button ] #Button when 'Button' Types is mounted should render 1`] = `
<DocumentFragment>
  <button
    class="button type-primary"
  >
    <h1
      class="heading color-inverted size-md weight-bold is-centered"
    >
      You Shall Not Pass!
    </h1>
  </button>
  <button
    class="button type-secondary"
  >
    <h1
      class="heading color-primary size-md weight-bold is-centered"
    >
      You Shall Not Pass!
    </h1>
  </button>
  <button
    class="button type-tertiary is-inline"
  >
    <h1
      class="heading color-primary size-md weight-bold is-centered"
    >
      You Shall Not Pass!
    </h1>
  </button>
</DocumentFragment>
`;
